cmd/host_logs_boots.go

```diff
@@ -39,3 +40,31 @@ Show all values that can be used with the boot arg to find logs.
 func init() {
 	hostLogsCmd.AddCommand(hostLogsBootsCmd)
 }
+
+func hostBootCompletions(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	if len(args) != 0 {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+	resp, err := helper.GenericJSONGet("host/logs/boots", "")
+	if err != nil || !resp.IsSuccess() {
+		return nil, cobra.ShellCompDirectiveNoFileComp
+	}
+
+	var ret []string
+	data := resp.Result().(*helper.Response)
+	if data.Result == "ok" && data.Data["boots"] != nil {
+		if boots, ok := data.Data["boots"].(map[string]interface{}); ok {
+			for bootID, bootName := range boots {
+				s := bootName.(string)
+				if toComplete == "" || strings.HasPrefix(s, toComplete) {
+					ret = append(ret, s)
+				}
+				if toComplete == "" || strings.HasPrefix(bootID, toComplete) {
+					ret = append(ret, bootID)
+				}
```

Comment:
So this works. It seems a little odd to me that it'll go like this:
```
<boot hash>
0
<boot hash>
-1
...
```
Rather then this:
```
<boot hash>
<boot hash>
...
0
-1
...
```
But probably not worth the effort.

The only thing that might be worth the effort is you can add description to items. Like if you look at addon completions [here](https://github.com/home-assistant/cli/blob/713510d4333b0bb25d8d0b677d5369bd73861f7c/cmd/addons.go#L46). You can see how after the slug we provide the name and url for context. I think those boot hashes probably need the bootID prefix as context since just a random boot hash isn't particularly useful.

Maybe it would be better to just prompt bootIDs with the offsets as details or vice versa? I know then we aren't completing all the possible valid values but I think users using the completion really just to be able to pick any boot. Having 2 items per boot probably doesn't help them.

Reply:
Hmmm :thinking: At first I wanted to add just the hashes, as they are more annoying to write/paste. But then I realized that the offsets would be more common (in fact, I never thought of using the hashes, I always only used the offset) and having them completed makes the completion... complete :sweat_smile: 

I'll have a look how to improve the UX of that, but if we should pick only one, then I'll hide the offsets and complete only hashes. But having both side-by-side in completion also has something to it...

In regard to:

> since just a random boot hash isn't particularly useful.

I consider the completion as a tool to make the task of writing something less annoying - so when I remember that the hash went like `68ed...`, I type `6` and then tab and add more tokens until I get the right value. So I disagree here - completing a random hash still might be useful.

Reply:
Agreed. With the descriptions (didn't know about them before) it also makes it even more pointless to complete it. The completion looks like this now:
```
db32e85e327a49bcbd05f4f3ea09eedf	boot offset -3
b3ed8ad3f86840778910e816cdad7167	boot offset -2
22100f35cd594bf78d0d574cb403feb1	boot offset -1
9abe1d6d7b004134993e40a8e37d0e36	boot offset 0
73cd933be6164cada777f4f48f5dc10a	boot offset -7
2b76a0fb7e354f859ed9b88ff86f2865	boot offset -6
84752b1466be48608b3ef4105c923b09	boot offset -5
4beae2e05fac4b3893a3649833d32f64	boot offset -4
```

The order changes every time, because Go doesn't guarantee order of iteration over maps. Are we fine with that?

Reply:
Ideally I guess it would be ordered by boot offset, but then, doesn't matter all that much to me :man_shrugging:

Reply:
Yeah it seems a pain to change that. Let's go with what we have now.